### PR TITLE
KAFKA-9595: switch usage of `alterConfigs` to `incrementalAlterConfigs` for kafka-configs tool

### PR DESCRIPTION
**Pull request by agam**
- Also, some minor refactoring for common code
- Test changes to `ConfigCommandTest`
- Verifies builds, tests pass

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by ijuma** [comment]
Do we fallback if incremental alter configs is not supported by the kafka broker version?

**Comment by agam** [comment]
As discussed, will pause this until resolving the right way to proceed with backward compatibility (will write a KIP), the options for handling older brokers are

1. explicitly don't support
2. automatically fallback
3. manually fallback (require passing an additional flag).

Will ping back here if/when that decision is reached.

File: core/src/main/scala/kafka/admin/ConfigCommand.scala [hunk 1]
```diff
@@ -307,42 +307,40 @@ object ConfigCommand extends Config {
...
         val newEntries = oldConfig ++ configsToBeAdded -- configsToBeDeleted
         val sensitiveEntries = newEntries.filter(_._2.value == null)
         if (sensitiveEntries.nonEmpty)
           throw new InvalidConfigurationException(s"All sensitive broker config entries must be specified for --alter, missing entries: ${sensitiveEntries.keySet}")
-        val newConfig = new JConfig(newEntries.asJava.values)
 
         val configResource = new ConfigResource(ConfigResource.Type.BROKER, entityName)
-        val alterOptions = new AlterConfigsOptions().timeoutMs(30000).validateOnly(false)
-        adminClient.alterConfigs(Map(configResource -> newConfig).asJava, alterOptions).all().get(60, TimeUnit.SECONDS)
+        adminClient.incrementalAlterConfigs(Map(configResource -> alterEntries).asJava, alterOptions).all().get(60, TimeUnit.SECONDS)
```
Comment (chia7712) [inline core/src/main/scala/kafka/admin/ConfigCommand.scala:343]: Could we use [Admin.incrementalAlterConfigs(Map)](https://github.com/apache/kafka/blob/trunk/clients/src/main/java/org/apache/kafka/clients/admin/Admin.java#L407)? the default AlterConfigsOptions should be fine to this tool.

**Comment by hachikuji** [comment]
@agam Going to close this for now. We can reopen once we are ready to pick it back up.

**Pull request closed without merging**